component/loki/source/gcplog/internal/gcplogtarget/types.go

```diff
@@ -21,22 +22,14 @@ type PullConfig struct {
 
 // PushConfig configures a GCPLog target with the 'push' strategy.
 type PushConfig struct {
-	HTTPListenAddress    string            `river:"http_listen_address,attr,optional"`
-	HTTPListenPort       int               `river:"http_listen_port,attr,optional"`
-	PushTimeout          time.Duration     `river:"push_timeout,attr,optional"`
-	Labels               map[string]string `river:"labels,attr,optional"`
-	UseIncomingTimestamp bool              `river:"use_incoming_timestamp,attr,optional"`
-}
-
-// DefaultPushConfig sets the default listen address and port.
-var DefaultPushConfig = PushConfig{
-	HTTPListenAddress: "0.0.0.0",
-	HTTPListenPort:    8080,
+	Server               *lhttp.ServerConfig `river:"server,block,optional"`
```

Comment:
Right now, users have to go one level deeper just to access the `server` block
```
loki.source.gcplog "foo" {
  server { 
    http { }
    grpc { }
  }
}
```



We have the option of _squashing` the block ([here's](https://github.com/grafana/agent/blob/bec518aa9ce7913debc20e88b9d458309e9f31b0/component/prometheus/remotewrite/types.go#L77) an example) so that people can configure the two inner blocks directly, like this:
```
loki.source.gcplog "foo" {
  http { }
  grpc { }
}
```


Code-wise, you still keep using the `PushConfig.Server` block as-is, this is just syntactic sugar for people writing configuration.

Reply:
Uh like it! Using squash since a three level deep configurations looks annoying.